ott/geometry/costs.py

```diff
@@ -98,7 +134,11 @@ def pairwise(self, x: jnp.ndarray, y: jnp.ndarray) -> float:
 
   def barycenter(self, weights: jnp.ndarray, xs: jnp.ndarray) -> float:
     """Output barycenter of vectors when using squared-Euclidean distance."""
-    return jnp.average(xs, weights=weights, axis=0)
+    return compute_weighted_mean(xs, weights)
+
+  @classmethod
```

Comment:
if added to the `CostFn` as is, not needed here.

Reply:
you mean you would suggest to revert back to ``jnp.average(xs, weights=weights, axis=0)``?